File: level_2/연속된 부분 수열의 합/solve.java
class Solution {
    public int[] solution(int[] sequence, int k) {
        int[] answer = new int[2];
        
//      투 포인터 선언
        int st = 0;
        int end = 0;
        
        int sum = 0;
        int size = sequence.length;
        
        for(end = 0; end < sequence.length; end++) {
            sum += sequence[end];
            
            while(end < sequence.length && sum > k) {
                sum -= sequence[st];
                st++;
            }
            
            if(sum == k && size > end - st) {
                size = end - st;
                answer[0] = st;
                answer[1] = end;
            }
        }

        return answer;
    }
}

File: level_2/연속된_부분_수열의_합/solve.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

// sequence_len은 배열 sequence의 길이입니다.
int* solution(int sequence[], size_t sequence_len, int k) {
    // return 값은 malloc 등 동적 할당을 사용해주세요. 할당 길이는 상황에 맞게 변경해주세요.
    int* answer = (int*)malloc(2 * sizeof(int));
    
    int st = 0, end = 0;
    int sum = 0;
    // answer의 최소 길이 저장용
    size_t min_len = sequence + 1;
    
    for(end = 0; end < sequence_len; end++) {
        sum += sequence[end];
        
        while(st <= end && sum > k) { // sum이 k보다 클 경우 st를 이동하며 sum 줄이기
            sum -= sequence[st++];
        }
        
        if(sum == k && (end - st + 1) < min_len) {
            min_len = end - st + 1;
            answer[0] = st;
            answer[1] = end;
        }
    }
    
    return answer;
}